### PR TITLE
Fix `Cursor(Stream)` to avoid incomplete reads and possible exception

### LINKED ISSUE #3492: Cursor(Stream) bugs
**.NET Core Version:**
master

**Have you experienced this same bug with .NET Framework?:**
probably

**Problem description:**
As noted [here in a PR](https://github.com/dotnet/winforms/pull/3476#discussion_r443216116) the constrcutor of `Cursor` taking a `Stream` is broken

```cs
public Cursor(Stream stream)
{
    if (stream == null)
    {
        throw new ArgumentNullException(nameof(stream));
    }

    int length = checked((int)stream.Length);
    _cursorData = new byte[length];
    stream.Read(_cursorData, 0, length);
    LoadPicture(
        new Ole32.GPStream(new MemoryStream(_cursorData)),
        nameof(stream));
}
```

- if the stream position isn't at zero the buffer would be allocated too large (suboptimal)
- if the stream doesn't know its length this would throw even though it doesn't have to (bug)
- `Read` is allowed to read less then `length` bytes so cursor data may be incomplete (bug)

Users can work around those bugs by wrapping their stream, but really the code should do a `stream.CopyTo(MemoryStream)` to let it do the heavy lifting and be more compatible to passing in arbitrary streams.

Note the current code doesn't throw, it just over-allocates and then reads less bytes than requested, which is fine if the cursor data is still represented completely at the current position.

**Expected behavior:**
- streams which don't expose their length can still be read
- streams which provide data in chunks still can be loaded

**Minimal repro:**
provide a custom stream

### PR DESCRIPTION
**Pull request by elachlan**
Fixes #3492

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8367)

### CODE REVIEW

**Comment by elachlan** [comment]
CC: @weltkante

**Comment by weltkante** [comment]
I don't really remember why I didn't make a PR back then, as the fix itself is trivial, maybe because it was unclear how much unit testing was wanted. As far as I can tell this looks good 👍

**Review by lonitra — APPROVED** [review]
LGTM

File: src/System.Windows.Forms/src/System/Windows/Forms/Cursor.cs [hunk 1]
```diff
@@ -81,12 +81,12 @@ public Cursor(Type type, string resource)
         public Cursor(Stream stream)
         {
             ArgumentNullException.ThrowIfNull(stream);
-
-            int length = checked((int)stream.Length);
-            _cursorData = new byte[length];
-            stream.Read(_cursorData, 0, length);
+            MemoryStream memoryStream = new();
+            stream.CopyTo(memoryStream);
+            memoryStream.Position = 0;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/Cursor.cs:86]: Is Position reset necessary here? `ToArray` doesn't care. Or was it for its use on line 89?

Comment (elachlan) [reply src/System.Windows.Forms/src/System/Windows/Forms/Cursor.cs]: stream.CopyTo advances all streams involved. I have updated the code for more clarity.

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**